src/overlays/actors/ovl_En_Invisible_Ruppe/z_en_invisible_ruppe.c

```diff
@@ -30,25 +30,95 @@ const ActorInit En_Invisible_Ruppe_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80C25B50 = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_NO_PUSH | OC1_TYPE_PLAYER, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK4, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_NO_PUSH | OC1_TYPE_PLAYER,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+
+    {
+        ELEMTYPE_UNK4,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
+
     { 10, 30, 0, { 0, 0, 0 } },
 };
 
-#endif
+void func_80C258A0(EnInvisibleRuppe* this, GlobalContext* globalCtx) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 32.0f, 30.0f, 0.0f, 4);
+}
+
+void func_80C2590C(EnInvisibleRuppe* this, GlobalContext* globalCtx) {
+    if (this->collider.base.ocFlags1 & OC1_HIT) {
+        switch (this->actor.params & 3) {
+            case 0:
+                play_sound(NA_SE_SY_GET_RUPY);
+                Item_DropCollectible(globalCtx, &this->actor.world.pos, 0x8000 | ITEM00_RUPEE_GREEN);
+                break;
+
+            case 1:
+                play_sound(NA_SE_SY_GET_RUPY);
+                Item_DropCollectible(globalCtx, &this->actor.world.pos, 0x8000 | ITEM00_RUPEE_BLUE);
+                break;
+
+            case 2:
+                play_sound(NA_SE_SY_GET_RUPY);
+                Item_DropCollectible(globalCtx, &this->actor.world.pos, 0x8000 | ITEM00_RUPEE_RED);
+                break;
+        }
+
+        if (this->unk_190 >= 0) {
+            Actor_SetSwitchFlag(globalCtx, this->unk_190);
+        }
+
+        this->actionFunc = func_80C259E8;
+    }
+}
+
+void func_80C259E8(EnInvisibleRuppe* this, GlobalContext* globalCtx) {
+    Actor_MarkForDeath(&this->actor);
+}
+
+void EnInvisibleRuppe_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnInvisibleRuppe* this = THIS;
+    s32 pad;
```

Comment:
Usually the pad goes above the actor recast (in cases that matter, it seems to always be above).